### ISSUE TITLE
[viki] Show better error for DRM

### ISSUE THREAD

**Issue by snowywish** [yt-dlp/yt-dlp#1379]
### Checklist

- [X] I'm reporting a site feature request
- [X] I've verified that I'm running yt-dlp version **2021.10.10**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

korea

### Description

whenever i try to download this show i keep getting this error
pls help

C:\Users\Muhammad Hayat>yt-dlp --username abc --password dbc https://www.viki.com/videos/1175019v-something-happened-in-bali-episode-5 --write-sub --sub-format srt --sub-lang en
[viki] Logging in
[viki] 1175019v: Downloading video JSON
[viki] 1175019v: Downloading video streams JSON
ERROR: [viki] 1175019v: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by <HTTPError 404: 'Not Found'>); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

**Comment by gamer191** [comment]
What's to say that yt-dlp isn't correctly detecting DRM? It looks more to me like it's just Snowywish using a blocked VPN, since they said that they are using a vpn and their verbose log reads:
`ERROR: [viki] 1175019v: Sorry, this content is not available in your region.`
(of note, their initial log gave a different 404 error)

NOTE: #3021 was closed as a duplicate of this issue, and #3021 might actually be caused by DRM

**Comment by nyuszika7h** [comment]
I think it's correctly detecting it now after the latest DRM bypass was patched, but the error message was far more vague before with the old API key/params.

**Comment by pukkandan** [comment]
> I think it's correctly detecting it now after the latest DRM bypass was patched

Could you show the log?

**Comment by nyuszika7h** [comment]
OK, the current situation is actually stranger than I thought:

<details>
<summary>Something Happened in Bali E05 – git master :x:</summary>

```
[debug] Command-line config: ['--ignore-config', '-v', '-F', 'https://www.viki.com/videos/1175019v-something-happened-in-bali-episode-5', '--username', 'PRIVATE',
 '--password', 'PRIVATE']                                                        
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8                                                                                       
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Lazy loading extractors is disabled                                                                                                                       
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 34baa9fdf                                                                                                                                       
[debug] Python version 3.10.1 (CPython 64bit) - Linux-4.19.0-18-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 5.0-static (setts), ffprobe 5.0-static, phantomjs .                                                                                  
[debug] Optional libraries: sqlite
[debug] Proxy map: {'http': 'REDACTED', 'https': 'REDACTED'}
[viki] Logging in                                                                
[debug] [viki] Extracting URL: https://www.viki.com/videos/1175019v-something-happened-in-bali-episode-5
[viki] 1175019v: Downloading video JSON                                                                                                                           
[viki] 1175019v: Downloading video streams JSON
[viki] 1175019v: Downloading initial MPD manifest
[viki] 1175019v: Downloading MPD manifest
ERROR: [viki] 1175019v: Failed to download MPD manifest: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://
github.com/yt-dlp/yt-dlp , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 638, in extract
    ie_result = self._real_extract(url)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/viki.py", line 267, in _real_extract
    formats = self._extract_mpd_formats(mpd_url, video_id)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 2666, in _extract_mpd_formats
    fmts, subs = self._extract_mpd_formats_and_subtitles(*args, **kwargs)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 2674, in _extract_mpd_formats_and_subtitles
    res = self._download_xml_handle(
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 955, in _download_xml_handle
    res = self._download_webpage_handle( 
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 797, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 782, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/extractor/common.py", line 764, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 3600, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/home/nyuszika7h/.pyenv/versions/3.10.1/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/home/nyuszika7h/.pyenv/versions/3.10.1/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/home/nyuszika7h/.pyenv/versions/3.10.1/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/home/nyuszika7h/.pyenv/versions/3.10.1/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/home/nyuszika7h/.pyenv/versions/3.10.1/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

</details>

<details>
<summary>Something Happened in Bali E05 – after removing mpdhd_high trick :white_check_mark:</summary>

```
[debug] Command-line config: ['--ignore-config', '-v', '-F', 'https://www.viki.com/videos/1175019v-something-happened-in-bali-episode-5', '--username', 'PRIVATE', '--password', 'PRIVATE']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 34baa9fdf
[debug] Python version 3.10.1 (CPython 64bit) - Linux-4.19.0-18-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 5.0-static (setts), ffprobe 5.0-static, phantomjs .
[debug] Optional libraries: sqlite
[debug] Proxy map: {'http': 'REDACTED', 'https': 'REDACTED'}
[viki] Logging in
[debug] [viki] Extracting URL: https://www.viki.com/videos/1175019v-something-happened-in-bali-episode-5
[viki] 1175019v: Downloading video JSON
[viki] 1175019v: Downloading video streams JSON
[viki] 1175019v: Downloading initial MPD manifest
[viki] 1175019v: Downloading MPD manifest
WARNING: [viki] Ignoring subtitle tracks found in the DASH manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: [viki] 1175019v: This video is DRM protected
Traceback (most recent call last):
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1399, in wrapper
    return func(self, *args, **kwargs)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1483, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1539, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 2470, in process_video_result
    self.raise_no_formats(info_dict)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 979, in raise_no_formats
    raise ExtractorError(msg, video_id=info['id'], ie=info['extractor'],
yt_dlp.utils.ExtractorError: [viki] 1175019v: This video is DRM protected
```

</details>

<details>
<summary>The One and Only E01 – whether I remove the mpdhd_high trick or not :x:</summary>

```
[debug] Command-line config: ['-v', '--ignore-config', '--playlist-items', '1', 'https://www.viki.com/tv/38264c-the-one-and-only']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8                                                                                       
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 34baa9fdf
[debug] Python version 3.10.1 (CPython 64bit) - Linux-4.19.0-18-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 5.0-static (setts), ffprobe 5.0-static, phantomjs .
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [viki:channel] Extracting URL: https://www.viki.com/tv/38264c-the-one-and-only
[viki:channel] 38264c: Downloading channel JSON
[download] Downloading playlist: The One and Only
[viki:channel] 38264c: Downloading Episodes JSON page 1
[viki:channel] playlist The One and Only: Downloading 1 videos
[download] Downloading video 1 of 1
[debug] [viki] Extracting URL: https://www.viki.com/videos/1190310v
[viki] 1190310v: Downloading video JSON
[viki] 1190310v: Downloading video streams JSON
[viki] 1190310v: Downloading initial MPD manifest
[viki] 1190310v: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, ae
xt, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
ERROR: [viki] 1190310v: Requested format is not available
Traceback (most recent call last):
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1399, in wrapper
    return func(self, *args, **kwargs)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1483, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 1539, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/nyuszika7h/src/yt-dlp/yt_dlp/YoutubeDL.py", line 2631, in process_video_result
    raise ExtractorError('Requested format is not available', expected=True,
yt_dlp.utils.ExtractorError: [viki] 1190310v: Requested format is not available

[download] Finished downloading playlist: The One and Only
```

</details>

<details>
<summary>The One and Only E01 – with `--allow-unplayable-formats` :white_check_mark:</summary>

```
[debug] Command-line config: ['-v', '--ignore-config', '--allow-unplayable-formats', '--test', '--playlist-items', '1', 'https://www.viki.com/tv/38264c-the-one-an
d-only']                                                                         
WARNING: You have asked for UNPLAYABLE formats to be listed/downloaded. This is a developer option intended for debugging.                                        
         If you experience any issues while using this option, DO NOT open a bug report
WARNING: --fixup is ignored since --allow-unplayable-formats was given
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 34baa9fdf
[debug] Python version 3.10.1 (CPython 64bit) - Linux-4.19.0-18-amd64-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 5.0-static (setts), ffprobe 5.0-static, phantomjs .
[debug] Optional libraries: sqlite
[debug] Proxy map: {'no': '.aiv-cdn.net,.media.dssott.com,akamaicdns3.hbogo.eu,edgecastcdns3.hbogo.eu,level3cdns.hbogo.eu,level3cdns3.hbogo.eu,.hulustream.com,wid
evinesucks'}
[debug] [viki:channel] Extracting URL: https://www.viki.com/tv/38264c-the-one-and-only
[viki:channel] 38264c: Downloading channel JSON
[download] Downloading playlist: The One and Only
[viki:channel] 38264c: Downloading Episodes JSON page 1
[viki:channel] playlist The One and Only: Downloading 1 videos
[download] Downloading video 1 of 1
[debug] [viki] Extracting URL: https://www.viki.com/videos/1190310v
[viki] 1190310v: Downloading video JSON
[viki] 1190310v: Downloading video streams JSON
[viki] 1190310v: Downloading initial MPD manifest
[viki] 1190310v: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, ae
xt, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1190310v: Downloading 1 format(s): 0+9
WARNING: You have requested merging of multiple formats while also allowing unplayable formats to be downloaded. The formats won't be merged to prevent data corru
ption.
[debug] Invoking downloader on "https://m-content6-viki.s.llnwi.net/1190310v/dash/1190310v_dash_high_1080p_drm_0cd8aa_2112221100-video-avc1.mp4"
[download] Destination: The One and Only - Episode 1 [1190310v].f0.mp4
[download] 100% of 10.00KiB in 00:00
[debug] Invoking downloader on "https://m-content6-viki.s.llnwi.net/1190310v/dash/1190310v_dash_high_720p_drm_1fa58a_2112221050-audio-en-mp4a.mp4"
[download] Destination: The One and Only - Episode 1 [1190310v].f9.m4a
[download] 100% of 10.00KiB in 00:00
[download] Finished downloading playlist: The One and Only
```

</details>

In the case of Something Happened in Bali (which has always been DRM-protected), it appears the embedded MPD URL still has a signature, which is why changing it invalidates it, resulting in a 403 error. We could check to make sure there are no query parameters before applying the workaround, or just remove it altogether since they seem to be forcing DRM on all videos once again, at least with the current parameters.

The One and Only didn't originally have DRM enforced, but I don't know what's going on with the "Requested format not available" there. The manifest definitely has a `ContentProtection` tag so I don't know why it's not throwing DRM protected error.

**Comment by pukkandan** [comment]
```
❯ yt-dlp -F https://www.viki.com/tv/38264c-the-one-and-only
[viki:channel] 38264c: Downloading channel JSON
[download] Downloading playlist: The One and Only
[viki:channel] 38264c: Downloading Episodes JSON page 1
[viki:channel] 38264c: Downloading Movies JSON page 1
[viki:channel] 38264c: Downloading Clips JSON page 1
[viki:channel] 38264c: Downloading Trailers JSON page 1
[viki:channel] playlist The One and Only: Downloading 21 videos
[download] Downloading video 1 of 21
[viki] 1190310v: Downloading video JSON
[viki] 1190310v: Downloading video streams JSON
[viki] 1190310v: Downloading initial MPD manifest
[viki] 1190310v: Downloading MPD manifest
[info] Available formats for 1190310v:
ID         EXT   RESOLUTION │ PROTO │ VCODEC MORE INFO
────────────────────────────────────────────────────────────────────
image_jpeg mhtml unknown    │ mhtml │ images DASH storyboards (jpeg)
```
🤣

**Comment by pukkandan** [comment]
> The One and Only E01 – whether I remove the mpdhd_high trick or not 

Fixed by c0b6e5c74d59d5260441ceaeb5dac3626780b31b

**Comment by nyuszika7h** [comment]
Made a PR ^ just in case we encounter non-DRM videos or another workaround in the future.
